### PR TITLE
feat: temporarily substitute alternative artifactory mirror

### PR DESCRIPTION
**Pull request by KyleAure**
Allow for transitional period between DockerHub and a self hosted solution.

- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

### CODE REVIEW

**Comment by KyleAure** [comment]
#build - personal build

**Comment by KyleAure** [comment]
#build - redo personal build after update
#libby - recheck headers after update

**Comment by KyleAure** [comment]
#build - redo personal build after update
#libby - recheck headers after update

**Comment by KyleAure** [comment]
#build - redo personal build after update
#libby - recheck headers after update

**Comment by KyleAure** [comment]
#build - redo personal build after update
#libby - recheck headers after update

**Comment by KyleAure** [comment]
Pipeline: passed
RTC Build: (not started yet)

Ready to merge once reviewed.

File: dev/fattest.simplicity/src/componenttest/containers/ArtifactoryImageNameSubstitutor.java
Comment (websterc87) [inline dev/fattest.simplicity/src/componenttest/containers/ArtifactoryImageNameSubstitutor.java:53]: Technically this is not a mirror so I would prefer if this said `tempRepository` and the variable below said
`tempRepositoryPrefixes`

But do not feel strongly enough to block this PR based on that.

File: dev/io.openliberty.org.testcontainers/cache/images
Comment (websterc87) [inline dev/io.openliberty.org.testcontainers/cache/images:2]: The changes in this file will nicely work with our automation.

**Review by websterc87 — APPROVED** [review]
Approved as the changes appear to have resulted in a successful PB and will cache/images will work nicely for our automation.

If this was due to last long term I would have more maintenance concerns but the change is good enough for the time it needs to last for.

Im not 100% versed in how the Name Substitutor works but the passing build suggest it works well enough.

**Pull request merged**
